src/data_access/FirebaseUserDataAccessObject.java

```diff
@@ -78,32 +78,35 @@ private String getAccessToken(String email, String password) {
   }
 
   @Override
-  public User get(String email, String password) {
+  public User getUser(String email, String password) {
     String idToken = getAccessToken(email, password);
     // Initialize user from our idToken and password by making a second call to Firebase
-    return getUserData(idToken, password);
+    return getPrivateUserData(idToken, password);
   }
 
-  public String getDisplayName(String uid) {
+  public DisplayUser getDisplayUser(String email) {
     // Note: This method doesn't require authentication, as anyone is allowed to retrieve a display
     // name given an uid
-    String url = String.format(Constants.DISPLAY_NAME_URL, uid);
+    // In addition, this method returns a RuntimeError if there is no DisplayUser available
+    String encodedEmail = Base64.getEncoder().encodeToString(email.toLowerCase().getBytes());
+    String url = String.format(Constants.DISPLAY_NAME_URL, encodedEmail);
     Request request = new Request.Builder().url(url).get().build();
 
     try {
       Response response = client.newCall(request).execute();
       if (response.isSuccessful()) {
-        return response.body().string().replace('"', ' ').trim();
+        String displayName = response.body().string().replace('"', ' ').trim();
+        return new DisplayUser(email, displayName);
       } else {
         throw new IOException();
       }
     } catch (IOException | JSONException e) {
-      throw new RuntimeException("Unable to retrieve display name. Please try again.");
+      throw new RuntimeException("Unable to retrieve user data. Please try again.");
     }
   }
 
-  // Return username, time user was created, and uid
-  private User getUserData(String idToken, String password) {
+  // Return time user was created and uid
```

Comment:
Prefer using a javadoc here

Reply:
We should standardize this when we add our official Javadoc to our codebase